packages/blue-sdk/src/token/Eip5267Domain.ts

```diff
@@ -0,0 +1,105 @@
+import type { Address } from "../types.js";
+
+export const EIP_712_FIELDS = [
+  "name",
+  "version",
+  "chainId",
+  "verifyingContract",
+  "salt",
+] as const;
+
+export type Eip712Field = (typeof EIP_712_FIELDS)[number];
+
+export interface IEip5267Domain {
+  fields: `0x${string}`;
+  name: string;
+  version: string;
+  chainId: bigint;
+  verifyingContract: Address;
+  salt: `0x${string}`;
+  extensions: readonly bigint[];
+}
+
+export class Eip5267Domain implements IEip5267Domain {
+  /**
+   * A bit map where bit i is set to 1 if and only if domain field i is present (0 ≤ i ≤ 4).
+   * Bits are read from least significant to most significant, and fields are indexed in the order that is specified by EIP-712, identical to the order in which they are listed in the function type.
+   */
+  public readonly fields;
+
+  /**
+   * The user readable name of signing domain, i.e. the name of the DApp or the protocol.
```

Comment:
this is more likely to be the name of the contract no?

Reply:
Correct, by there's no "name" on a contract per-se and I simply copied EIP712 docs here ^^